config/initializers/github_client.rb

```diff
@@ -2,10 +2,24 @@ module GitHubClassroom
   def self.github_client(options = {})
     client_options = {
       client_id:     Rails.application.secrets.github_client_id,
-      client_secret: Rails.application.secrets.github_client_secret,
-      auto_paginate: true
+      client_secret: Rails.application.secrets.github_client_secret
     }.merge!(options)
 
+    if client_options.has_key?(:access_token)
```

Comment:
This is getting a bit hairy with all of these `merge!` calls, let's change it to:

```ruby
def self.github_client(options = {})
  client_options = {}.tap do |opts|
    opts[:user_agent]    = "GitHub Classroom" # This is new, but I want to add it 😄
    opts[:auto_paginate] = true # I want to put this back so I added it here.

    if options.has_key?(:access_token)
      opts[:access_token] = options[:access_token]
    else
      options[:client_id]     = Rails.application.secrets.github_client_id
      options[:client_secret] = Rails.application.secrets.github_client_secret
    end
    
    if enterprise_instance?
      api_endpoint = "https://#{Rails.application.secrets.github_enterprise_hostname}/api/v3"
      opts[:api_endpoint] = api_endpoint
    end

    Octokit::Client.new(client_options)
  end
end